docs/developer-docs/functionality/http_requests/http_requests-how-it-works.md

```diff
@@ -0,0 +1,152 @@
+* The HTTP response from the server is received by the adapter on each replica of the subnet. The adapter limits the response size to `max_response_bytes` which defaults to $2$ MB and can be set to lower values. It is important to set this as low as reasonably possible for the expected response as it affects the price of the request.
+* An optional transformation function implemented as part of the canister is invoked on the response on each replica to transform it.
+* The transformed response is handed over to consensus on each replica.
+* Consensus agrees on a response if at least $2/3$ of the replicas have the same response for the request as input. In this case, consensus provides this response back to the management canister API at the execution layer, or an error if no consensus can be reached or in case of other problems.
+* The execution layer provides the response or error back to the calling canister.
+
+As we can see, the transformation function and IC consensus play a crucial role here: The transformation function ensures that differences in the responses received by the replicas are removed and transformed responses on different replicas will be exactly the same. thus enabling consensus to provide the agreed-upon response to the calling canister.
+
+The trust model for canister HTTP outcalls is a combination of the trust model of the called HTTP server and the IC:
+* It is assumed that the HTTP service is honest, otherwise, it can provide any responses to any of the calling replicas of the subnet. This is the case also for calls to the service from a Web 2.0 service or an oracle, thus, the situation is the same as in our scenario of a blockchain making the call to the service.
+* The trust model of the IC assumes that at least $2/3$ of the replicas are honest. In this case, the honest replicas will obtain the same response (after optional transformation) and consensus will be able to agree on the response so that the response can be provided back to the calling canister.
+A dishonest server can easily make requests fail or provide wrong data. Also, $1/3$ or more of dishonest subnet replicas can make requests fail. For providing wrong data, more than $2/3$ of the replicas need to be compromised.
+
+The programming model of this feature is such that the canister making an HTTP outcall acts as *HTTP client*, i.e., needs to, in the general case, interpret various headers and act accordingly. For use cases such as API requests this is rather straightforward and does not require many specific considerations in the standard case. The IC protocol stack can conceptually be simply seen as a communication pipe between the canister and the conventional HTTP server that makes sure that the HTTP response makes it through consensus and all honest replicas receive the exact same response in execution, i.e., the calling canister receives an agreed-upon response.
```

Comment:
talking about `outcall` again. Google doesn't know this term almost at all.

Reply:
HTTPS outcalls is the terminology used in our Wiki page on the topic.